cli/src/cmd/forge/test/mod.rs

```diff
@@ -169,54 +174,52 @@ impl TestArgs {
             .build(project.paths.root, output, env, evm_opts)?;
 
         if self.debug.is_some() {
-            filter.args_mut().test_pattern = self.debug;
-
-            match runner.count_filtered_tests(&filter) {
-                1 => {
-                    // Run the test
-                    let results = runner.test(&filter, None, test_options)?;
-
-                    // Get the result of the single test
-                    let (id, sig, test_kind, counterexample) = results.iter().map(|(id, SuiteResult{ test_results, .. })| {
-                        let (sig, result) = test_results.iter().next().unwrap();
-
-                        (id.clone(), sig.clone(), result.kind.clone(), result.counterexample.clone())
-                    }).next().unwrap();
-
-                    // Build debugger args if this is a fuzz test
-                    let sig = match test_kind {
-                        TestKind::Fuzz { first_case, .. } => {
-                            if let Some(CounterExample::Single(counterexample)) = counterexample {
-                                counterexample.calldata.to_string()
-                            } else {
-                                first_case.calldata.to_string()
-                            }
-                        },
-                        _ => sig,
-                    };
-
-                    // Run the debugger
-                    let mut opts = self.opts.clone();
-                    opts.silent = true;
-                    let debugger = DebugArgs {
-                        path: PathBuf::from(runner.source_paths.get(&id).unwrap()),
-                        target_contract: Some(get_contract_name(&id).to_string()),
-                        sig,
-                        args: Vec::new(),
-                        debug: true,
-                        opts,
-                        evm_opts: self.evm_opts,
-                    };
-                    utils::block_on(debugger.debug())?;
-
-                    Ok(TestOutcome::new(results, self.allow_failure))
+            filter.args_mut().test_pattern = self.debug.clone();
+
+            let filtered_tests = runner.get_filtered_sig(&filter);
+            // filter any fuzzed test
+            let mut filtered_tests = filtered_tests
+                .into_iter()
+                .filter(|test| {
+                    let mut chars = test.chars();
+                    chars.position(|c| c == '(').unwrap();
+                    chars.next() == Some(')')
+                })
+                .collect::<Vec<_>>();
+
+            let old_len = filtered_tests.len();
+
+            filtered_tests.sort();
+            filtered_tests.dedup();
+
+            if old_len != filtered_tests.len() {
+                return Err(eyre::eyre!("There was some duplicate test function names."))
+            }
+
+            let test_name = match filtered_tests.len() {
+                // or should we display them all ?
```

Comment:
Sorry what's the question here exactly? Is the choice between (1) always showing ALL tests vs. (2) allowing you to filter the ones shown? If so, supporting both seems reasonable to me if that's feasible? It'd be nice to do something like `forge test --debug Unauthorized` and get a list of all test names that contain `Unauthorized`. Sometimes it can be tedious to select a test when you have the same test name in multiple contracts, so this can helper 

(But maybe I am misunderstanding the question here)